src/main/java/edu/ucsb/cs156/courses/repositories/PersonalSectionsRepository.java

```diff
@@ -0,0 +1,19 @@
+package edu.ucsb.cs156.courses.repositories;
+
+import edu.ucsb.cs156.courses.models.PersonalSection;
+import edu.ucsb.cs156.courses.entities.User;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+
+@Repository
+public interface PersonalSectionsRepository extends CrudRepository<PersonalSection, Long> {
+  Optional<PersonalSection> findByPsId(Long psId);
+  Optional<PersonalSection> findByIdAndUser(long id, User user);
+
+  Iterable<PersonalSection> findAllByPsId(Long psId);
+  Iterable<PersonalSection> findAllByPsIdAndUser(Long psId, User user);
+  Iterable<PersonalSection> findAllByUserId(Long user_id);
+}
```

Comment:
Yes, we need a PersonalSections model.. but I don't think we need a PersonalSectionsRepository.

What is your intention for this?  It seems as if it would be storing redundant data, which is a bad idea.

The idea of a PersonalSection model is that we would populate it and then return it from an endpoint, but we would populate it from the existing database tables and/or the UCSB API.   The intention is NOT to create a new database.